libs/agent-sm/agent/src/main/java/org/opensearch/javaagent/Agent.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.javaagent;
+
+import org.opensearch.javaagent.bootstrap.AgentPolicy;
+
+import java.lang.instrument.Instrumentation;
+import java.nio.channels.SocketChannel;
```

Comment:
are all Socket calls in Opensearch is contained within in the context the Nio Transport / Http channel?

Reply:
The majority of the Socket calls are performed by the plugins (security,  s3, hdfs, ml-commons, ....)

Reply:
👍